.github/workflows/main.yaml

```diff
@@ -24,20 +27,40 @@ jobs:
       - name: Analyze project source
         uses: invertase/github-action-dart-analyzer@v1
 
+      - uses: codecov/codecov-action@v2
+        with:
+          files: coverage/lcov.info
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        flutter-version: ['3.16.9', '3.19.6', '3.22.3', '3.24.3']
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: subosito/flutter-action@v2
+        with:
+          flutter-version: ${{ matrix.flutter-version }}
+          channel: 'stable'
+          cache: true
+
+      - name: Install Dependencies
+        run: flutter packages get
+
       - name: Run tests
-        run: flutter test --no-pub --coverage --test-randomize-ordering-seed=random
+        run: ALCHEMIST_FLUTTER_VERSION=${{ matrix.flutter-version }} flutter test --no-pub --coverage --test-randomize-ordering-seed=random
```

Comment:
Gonna look at allowing us to drop this env variable and instead grab it via `Process.run`. This will ensure that the version Alchemist uses for golden comparison is guaranteed to match the actual running version.

Reply:
`Process.run` sounds good to me, but be aware that:

1. If people run `fvm`, simply running a `flutter` command might give incorrect output.
2. Instead of doing some string voodoo, you could do
```dart
import 'dart:convert';

final output = Process.run('flutter --version --machine');
final flutterData = json.decode(output);
final flutterVersion = flutterData['flutterVersion'];
```

Reply:
Ohh, that's way better haha. I just pushed a commit that uses that; if everything succeeds, I think this is good to merge now 👀 

Also, 🎉 good news. I was able to explicitly set button colors in the interactions test which resolved the breaking changes that we were inheriting from flutter 3.22.x, meaning we don't need a separate directory for that version, everything is passing on 3.16.x and beyond 🎉